### PR TITLE
Feature 112 load idle page from api

### LINKED ISSUE #112: Changing idle page support
As an exhibition creator I would like to be able to change the look of the idle page.

### PR DESCRIPTION
**Pull request by MrWalrus3451**
Resolves #112

### CODE REVIEW

File: app/src/main/java/fi/metatavu/muisti/exhibitionui/views/MainActivity.kt [hunk 1]
```diff
@@ -118,9 +140,7 @@ class MainActivity : MuistiActivity() {
     }
 
     /**
-     * Checks if page is constructed in the PageViewContainer and either navigates to it or keeps waiting
-     *
-     * @param pageId page id to navigate to once it is ready
+     * Checks if forced portrait mode setting is loaded and modifies UI accordingly
```
Comment (MrWalrus3451) [inline app/src/main/java/fi/metatavu/muisti/exhibitionui/views/MainActivity.kt:143]:
```suggestion
     * Checks if forced portrait mode setting is loaded and implements it
```

File: app/src/main/java/fi/metatavu/muisti/exhibitionui/views/MainViewModel.kt [hunk 2]
```diff
@@ -27,4 +31,17 @@ class MainViewModel(application: Application): AndroidViewModel(application) {
     suspend fun getFrontPage(language: String) : UUID? {
         return pageRepository.findIndexPage(language = language)?.pageId
     }
+
+    /**
+     * Returns the idle page Id for the current selected device
```
Comment (MrWalrus3451) [inline app/src/main/java/fi/metatavu/muisti/exhibitionui/views/MainViewModel.kt:36]:
```suggestion
     * Returns the idle page Id for the current device
```

File: app/src/main/java/fi/metatavu/muisti/exhibitionui/views/MainActivity.kt [hunk 3]
```diff
@@ -24,16 +24,31 @@ class MainActivity : MuistiActivity() {
...
+            runOnUiThread {
+                if (idlePage != null) {
+                    setContentView(R.layout.activity_page)
+                    openIdlePage(idlePage)
+                } else {
+                    setContentView(R.layout.activity_main)
+                }
+                supportActionBar?.hide()
+                listenSettingsButton(settings_button)
+                //listenLoginButton(login_button)
```
Comment (anttileppa) [inline app/src/main/java/fi/metatavu/muisti/exhibitionui/views/MainActivity.kt:48]: Why this is commented out?

File: app/src/main/java/fi/metatavu/muisti/exhibitionui/views/MainActivity.kt [hunk 4]
```diff
@@ -56,6 +71,13 @@ class MainActivity : MuistiActivity() {
         ExhibitionUIApplication.instance.readApiValues()
     }
 
+    private fun openIdlePage(idlePage: PageView) {
```
Comment (anttileppa) [inline app/src/main/java/fi/metatavu/muisti/exhibitionui/views/MainActivity.kt:74]: Add docs

File: app/src/main/java/fi/metatavu/muisti/exhibitionui/views/MainViewModel.kt [hunk 5]
```diff
@@ -27,4 +31,17 @@ class MainViewModel(application: Application): AndroidViewModel(application) {
...
     }
-}
\ No newline at end of file
+
+    /**
+     * Returns the idle page Id for the current device
+     *
+     * @return a front page id for given language or null if not found
+     */
+    suspend fun getIdlePageId() : UUID? {
```
Comment (anttileppa) [inline app/src/main/java/fi/metatavu/muisti/exhibitionui/views/MainViewModel.kt:40]: Fix comments

**Review by anttileppa — CHANGES_REQUESTED** [review]

**Pull request merged**